### PR TITLE
Fix incorrect trash unicode address, and tweak build process

### PR DESCRIPTION
**Pull request by jasmussen**
This PR intends to address regressions found as part of the attempt to merge the new icon font (as built by the grunt script) into 4.9. The full thread is here: https://core.trac.wordpress.org/ticket/41074, but a condensed task list is the following:

1. [ ] Remove WOFF file from WordPress core — we now use WOFF2, and it's embedded in the file
2. [ ] Remove SVG file from WordPress core — we don't use it anymore, and the file generated by Dashicons is a sprite, not a font.
3. [x] Align CSS coding standard
4. [x] Fix char positions to match old ones (trash had the wrong placement)
5. [x] Extra CSS markup to be generated

This PR addresses 4, 5, and partially addresses 3. 1 and 2 have to happen upstream.

It now includes a CSS template file, so we can better tweak the spacing and layout of the files. But it looks like we can't edit `url("...")` to be `url( "..." )` without hacking core files.

I haven't found a way to adjust the order in which icons are output in the CSS file — it's done alphabetically. So I don't know that we can sort the CSS classes.

The trash icon issue seems to have been due to a file being renamed (see https://github.com/WordPress/dashicons/commit/8bd649b2e266f7e5578fdac34106754f1ecfcdba) without the codepoint also moving.

To add the extra markup, I edited the CSS template file. 

**So to summarize:** this PR fixes the bugs that came with the build process and makes the CSS files better adhere to WP standards. But it does not (can't) go all the way to be a 100% drop-in replacement. 

CC: @ntwb

### CODE REVIEW

**Comment by field2** [comment]
What order should the icons be in, if not alphabetically?

**Comment by jasmussen** [comment]
> What order should the icons be in, if not alphabetically?

I think just the same as the were, to make comparisons easier, see: https://core.trac.wordpress.org/timeline?from=2017-10-25T07%3A13%3A27%2B02%3A00&precision=second

**Comment by field2** [comment]
There are actually a lot of differences between this dashicons.css file and the one in the WP repo. Would it be better to just manually update the WP css file as we've done in the past, until we can figure out how to resolve the two?
